### ISSUE TITLE
Just Actually Use the user's git diff?

### ISSUE THREAD

**Issue by Gankra** [bholley/cargo-vet#3]
The book says we should emit "a git-compatible diff", and the easiest way I saw to do that was... just use `git diff --no-index`! While in general i don't like assuming the user has external tools, this is mildly justified for a few reasons:

* Everyone actually just has git
* We are invoked "through" cargo, and cargo depends on git because the crates.io index is git-based 
   * Cargo I believe uses the [git2](https://docs.rs/git2/latest/git2/) crate

I should look into whether the way cargo works *actually* requires git to be installed, or is itself basically just vendoring git-as-a-library. Basically I think it's sound to depend on whatever Cargo is depending on, and at worst we should just use git2. (Using Command::new("git") was just an easy way to Get It Working.)

**Comment by bholley** [comment]
Yes, I am fine with depending on git for the time being and circling back if it ever becomes a problem.
